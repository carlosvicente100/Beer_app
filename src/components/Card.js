import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const CardContainer = styled.div`
  background-color: #706fd3;
  color: black;
  border-radius: 25px 0 25px 0;
  overflow: hidden;
  height: 200px;
  display: grid;
  grid-template-rows: 50px 20px auto;
  grid-template-columns: 125px auto auto;
  column-gap: 10px;
  grid-template-areas:
    'image title title'
    'image . .'
    'image description description';
  img {
    grid-area: image;
  }
  h1 {
    height: auto;
    grid-area: title;
    font-size: 16px;
    text-transform: uppercase;
  }
  p {
    grid-area: description;
    font-size: 14px;
  }
`

const CardImage = styled.img`
  margin-inline-start: 20px;
  height: 200px;
  width: auto;
`

const Card = ({ name, description, image_url, tagline, first_brewed }) => {
  const checkLength = (string, maxLenght = 120) => {
    if (string.length > maxLenght) {
      return string.slice(0, maxLenght) + '...'
    } else {
      return string
    }
  }
  return (
    <CardContainer>
      <CardImage src={image_url}></CardImage>
      <h1>{name}</h1>
      <span>{`${tagline}  (${first_brewed})`}</span>
      <p>{checkLength(description)}</p>
    </CardContainer>
  )
}
Card.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image_url: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  first_brewed: PropTypes.string.isRequired
}

Card.defaultProps = {}

export default Card
