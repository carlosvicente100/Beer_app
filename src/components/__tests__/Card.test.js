import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Card from '../Card'

const dummyProps = {
  name: 'dummy Name',
  description: 'dummy description extra large for test shorter function if is required',
  image_url: 'http://placekitten.com/200/300',
  tagline: 'dummy tagline',
  first_brewed: '05/2020'
}

describe('Card Component Suite', () => {
  test('Snapshot check', () => {
    const component = renderer.create(<Card {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Should use props correcty', () => {
    const { getByText } = render(<Card {...dummyProps} />)
    expect(getByText(dummyProps.name)).toBeInTheDocument()
    expect(getByText(`${dummyProps.tagline} (${dummyProps.first_brewed})`)).toBeInTheDocument()
    expect(getByText(`${dummyProps.tagline} (${dummyProps.first_brewed})`)).toBeInTheDocument()
    expect(getByText(dummyProps.description)).toBeInTheDocument()
  })

  describe('Card Component Internal Functions', () => {
    test('checkLength Function', () => {
      const longDescription =
        "dummy description extra large for test shorter function if is required , for that reason, I'm extending this value with random content"
      const { getByText } = render(<Card {...dummyProps} description={longDescription} />)
      expect(getByText(/I'm extending this value with .../)).toBeInTheDocument()
    })
  })
})
