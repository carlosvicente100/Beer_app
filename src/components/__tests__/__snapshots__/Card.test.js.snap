// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Suite Snapshot check 1`] = `
.c0 {
  background-color: #706fd3;
  color: black;
  border-radius: 25px 0 25px 0;
  overflow: hidden;
  height: 200px;
  display: grid;
  grid-template-rows: 50px 20px auto;
  grid-template-columns: 125px auto auto;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  grid-template-areas: 'image title title' 'image . .' 'image description description';
}

.c0 img {
  grid-area: image;
}

.c0 h1 {
  height: auto;
  grid-area: title;
  font-size: 16px;
  text-transform: uppercase;
}

.c0 p {
  grid-area: description;
  font-size: 14px;
}

.c1 {
  margin-inline-start: 20px;
  height: 200px;
  width: auto;
}

<div
  className="c0"
>
  <img
    className="c1"
    src="http://placekitten.com/200/300"
  />
  <h1>
    dummy Name
  </h1>
  <span>
    dummy tagline  (05/2020)
  </span>
  <p>
    dummy description extra large for test shorter function if is required
  </p>
</div>
`;
